### PR TITLE
Update gemfile references to refer to legacy ruby branch

### PR DESCRIPTION
**Pull request by NickElliot**
Part 1 of ruby removals

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .ci/containers/build-environment/Dockerfile [hunk 1]
```diff
@@ -39,8 +39,8 @@ RUN git config --global user.email "magic-modules@google.com"
 RUN go install golang.org/x/tools/cmd/goimports@d088b475e3360caabc032aaee1dc66351d4e729a
 RUN go install github.com/github/hub@v2.11.2+incompatible
 
-ADD "https://raw.githubusercontent.com/GoogleCloudPlatform/magic-modules/main/mmv1/Gemfile" Gemfile
-ADD "https://raw.githubusercontent.com/GoogleCloudPlatform/magic-modules/main/mmv1/Gemfile.lock" Gemfile.lock
+ADD "https://raw.githubusercontent.com/GoogleCloudPlatform/magic-modules/refs/heads/legacy-ruby/mmv1/Gemfile" Gemfile
+ADD "https://raw.githubusercontent.com/GoogleCloudPlatform/magic-modules/refs/heads/legacy-ruby/mmv1/Gemfile.lock" Gemfile.lock
```
Comment (zli82016) [inline .ci/containers/build-environment/Dockerfile:43]: Have you tested the new Dockerfile locally to  make sure no errors occur, even though it is a minor change? 

I think this is the doc 
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/containers/README.md#updating-a-docker-image

**Comment by NickElliot** [comment]
Followed the testing steps mentioned and the testing image is referenced in the execution details.

Believe it would involve undoing the "testing" reference before this is merged, and pushing up the image to the base but will verify first

File: mmv1/third_party/terraform/services/compute/metadata.go.tmpl [hunk 2]
```diff
@@ -31,7 +31,7 @@ func MetadataRetryWrapper(update func() error) error {
 
 // Update the metadata (serverMD) according to the provided diff (oldMDMap v
 // newMDMap).
-func MetadataUpdate(oldMDMap map[string]interface{}, newMDMap map[string]interface{}, serverMD *compute.Metadata) {
+func MetadataUpdate(oldMDMapA map[string]interface{}, newMDMap map[string]interface{}, serverMD *compute.Metadata) {
```
[the remark this answers is not in the document]
Comment (NickElliot) [reply mmv1/third_party/terraform/services/compute/metadata.go.tmpl]: Just undid it, was to force the rebuilds

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**